### PR TITLE
Add new Rule: arrow-parens

### PR DESCRIPTION
**Pull request by YuichiNukiyama**
Impliment #777 
Arrow function allow ommit parentheses when parameter is single. This PR enforces parentheses around typeless parameter of arrow function .
This PR doent't warn typed parameter. Becase, TS don't allow ommit parentheses when we use typed parameter. following code, TS warn syntax error and TSLint don't

```
var a = b:number => {};
````

### CODE REVIEW

File: src/rules/arrowParensRule.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import * as ts from "typescript";
+
+import * as Lint from "../lint";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "arrow-parens",
+        description: "Requires parentheses when using arrow function.",
```
Comment (JKillian) [inline src/rules/arrowParensRule.ts]: Can we change this to: "Requires parentheses around the parameters of arrow function declarations."

File: README.md [hunk 2]
```diff
@@ -224,6 +224,7 @@ Core rules are included in the `tslint` package.
   * `"parameters"` checks alignment of function parameters.
   * `"arguments"` checks alignment of function call arguments.
   * `"statements"` checks alignment of statements.
+* `"arrow-parens"` requires parentheses when using arrow function.
```
Comment (JKillian) [inline README.md]: Can we change this to "requires parentheses around the parameters of arrow function declarations."

File: src/rules/arrowParensRule.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        rationale: "Maintains stylistic consistency with other arrow function definition.",
```
Comment (JKillian) [inline src/rules/arrowParensRule.ts]: "definitions" instead of "definition"

File: src/rules/arrowParensRule.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        optionsDescription: "Not configurable.",
+        options: null,
+        optionExamples: ["true"],
+        type: "style",
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_STRING = "Parentheses are required when define arrow function";
```
Comment (JKillian) [inline src/rules/arrowParensRule.ts]: "Parentheses are required around the parameters of an arrow function definition."

**Comment by JKillian** [comment]
Looks good @YuichiNukiyama! Mind making the small documentation tweaks mentioned above?

**Comment by YuichiNukiyama** [comment]
Thanks @JKillian. I did it.

**Comment by JKillian** [comment]
Thanks @YuichiNukiyama!

**Pull request merged**